src/server.ts

```diff
@@ -79,7 +85,8 @@ export default async ({
   await postRegisterHook.call(this, app);
 
   app.db = await dbConnect(config);
-  app.schema = schema(app.db);
+  const databaseClient = app.db as unknown;
```

Comment:
Why the cast here?

Reply:
dbConnect(config) returns void| import('mongoose') which differs from DbClient which is Promise<Mongoose> | SchemaBuilder. In the new PR I have casted this at the dbConnect layer to DbClient.